### PR TITLE
add basic examples

### PR DESCRIPTION
**Pull request by ttdonovan**
add a few basic examples and suggested "shared" for common wallet and anchor client setup

### CODE REVIEW

File: .env.sample
Comment (mindrunner) [inline .env.sample:1]: nice!

File: NOTES.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
+# Notes
+
+## Development Setup
+
+### Install Solana
+
+https://docs.solanalabs.com/cli/install
+
+- `sh -c "$(curl -sSfL https://release.solana.com/v1.18.4/install)"`
```
Comment (mindrunner) [inline NOTES.md:9]: I think `1.18.11` is out

File: examples/01-sage-game/index.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    const games = await readAllFromRPC(
+        connection,
+        sageProgram as any,
+        Game,
+        'processed',
+        [],
+    );
+
+    if (games.length === 0) {
+        throw 'no game found';
```
Comment (mindrunner) [inline examples/01-sage-game/index.ts:31]: It's better to `throw Error('no game found')`

catching exceptions is not type-safe, plus you loose information (e.g. the stack) if you throw strings

File: examples/04-sage-planets/index.ts [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+import { MineItem, Planet, Resource, SAGE_IDL } from "@staratlas/sage";
+
+import { newConnection, newAnchorProvider } from '../../shared/anchor-setup';
+import { generateKeypair } from '../../shared/wallet-setup';
+
+import { Table } from 'console-table-printer';
+
+const SAGE_PROGRAM_ID = 'SAGE2HAwep459SNq61LHvjxPk4pLPEJLoMETef7f7EE';
+
+const mainFunction = async (connection: Connection, myWallet: Keypair) => {
```
Comment (mindrunner) [inline examples/04-sage-planets/index.ts:15]: just `main` would rather be convention

**Review by mindrunner — CHANGES_REQUESTED** [review]
Nice, thanks for the PR. Added some comments.

File: NOTES.md (hunk 1 above)
Comment (ttdonovan) [reply NOTES.md:9]: I just copied what was in the other README.md with the project and moved it here... thinking that the dir/files `sage-player-profile` will be deleted... if you go here still says v1.18.4 https://docs.solanalabs.com/cli/install

File: examples/04-sage-planets/index.ts (hunk 3 above)
Comment (ttdonovan) [reply examples/04-sage-planets/index.ts]: See commit [7100121](https://github.com/staratlasmeta/star-atlas-cookbook/pull/1/commits/710012175c04d238db757ca99467af89103325c0)

File: examples/01-sage-game/index.ts (hunk 2 above)
Comment (ttdonovan) [reply examples/01-sage-game/index.ts]: See commit [7100121](https://github.com/staratlasmeta/star-atlas-cookbook/pull/1/commits/710012175c04d238db757ca99467af89103325c0)

File: NOTES.md (hunk 1 above)
Comment (mindrunner) [reply NOTES.md:9]: I think the docs are not super actively maintained. The most recent `1.18` version will hit mainnet in a couple of days (possible fix for the congested network). I think we can have our docs up to date already.

File: examples/04-sage-planets/index.ts (hunk 3 above)
Comment (mindrunner) [reply examples/04-sage-planets/index.ts]: > again just copied what was done in first example script but will change to `main`.

Yea I am aware. I promised @cuniculture a review of this initial code, so your PR was just perfectly fine to add my two cents. All minor comments anyways. 

Thanks for contribution.

**Review by mindrunner — APPROVED** [review]

File: NOTES.md (hunk 1 above)
Comment (mindrunner) [reply NOTES.md:9]: Ill merge. We can update the docs later.

**Pull request merged**